src/partners/ioniavisarewards.ts

```diff
@@ -0,0 +1,156 @@
+import {
+  asArray,
+  asMaybe,
+  asNumber,
+  asObject,
+  asString,
+  asUnknown,
+  asValue
+} from 'cleaners'
+
+import {
+  asStandardPluginParams,
+  PartnerPlugin,
+  PluginParams,
+  PluginResult,
+  StandardTx,
+  Status
+} from '../types'
+import { datelog, retryFetch, smartIsoDateFromTimestamp, snooze } from '../util'
+
+const asIoniaStatus = asMaybe(asValue('complete'), 'other')
+
+const asIoniaTx = asObject({
+  Id: asNumber,
+  CreatedDate: asString,
+  GiftCardFaceValue: asNumber,
+  USDPaidByCustomer: asNumber,
+  MerchantName: asString
+})
+
+const asIoniaResult = asObject({
+  Data: asObject({
+    Transactions: asArray(asUnknown)
+  })
+})
+
+type IoniaTx = ReturnType<typeof asIoniaTx>
+type IoniaStatus = ReturnType<typeof asIoniaStatus>
+
+const MAX_RETRIES = 5
+const LIMIT = 200
+const QUERY_LOOKBACK = 1000 * 60 * 60 * 24 * 5 // 5 days
+
+const statusMap: { [key in IoniaStatus]: Status } = {
+  complete: 'complete',
+  other: 'complete'
+}
```

Comment:
Same thing for this copy-pasted status stuff, which isn't needed.

Reply:
It’s really there as a template for when Ionia adds statuses. Their API is still being modified per our feedback and that’s one of the pending changes.

Reply:
Ok, I didn't have that bit of context. My preference is to keep things clean until they are needed, and then copy / paste statuses from another plugin, but I see how this way makes sense too. Merge away.